tests/suggestion-box.test.ts

```diff
@@ -0,0 +1,309 @@
+
+        el = await fixture(
+            html`<aalam-sgn-box
+                ><div slot="sgn-empty">
+                    This is the empty block
+                </div></aalam-sgn-box
+            >`
+        );
+        input = el.shadowRoot?.querySelector("input");
+        dd = el.shadowRoot?.querySelector("#sgn-container");
+        expect(dd?.style.display).to.equal("none");
+        input?.focus();
+        await el.updateComplete;
+        expect(dd?.style.display).to.equal("block");
+    });
+
+    it("should render #sgn-container", async () => {
+        const el = await fixture(html`<aalam-sgn-box></aalam-sgn-box>`);
+        const container = el.shadowRoot?.getElementById("sgn-container");
+        expect(container).to.exist;
+    });
+
+    it("should render #private-item", async () => {
+        const el = await fixture(html`<aalam-sgn-box></aalam-sgn-box>`);
+        const privateitem = el.shadowRoot?.getElementById("private-item");
+        expect(privateitem).to.exist;
+    });
+
+    it("should update suggestions on input event", async () => {
+        const el = await fixture(
+            html`
+                <aalam-sgn-box>
+                    <div slot="sgn-item-template">{name}</div>
+                </aalam-sgn-box>
+            `
+        );
+        const input = el.shadowRoot?.querySelector("input");
+        el.list = [
+            {
+                name: "thiruvarur",
+                description: "My native",
+            },
+            {
+                name: "coimbatore",
+                description: "Working place",
+            },
+            {
+                name: "chennai",
+                description: "dream location",
+            },
+        ];
+        el.listkey = "name";
+        input.value = "thi";
+        input.dispatchEvent(
+            new Event("input", { bubbles: true, composed: true })
+        );
+        await el.updateComplete;
+
+        expect(el.filtered_list.length).to.equal(1);
+        expect(el.filtered_list[0].name).to.equal("thiruvarur");
+    });
+
+    it("should update suggestions using setSuggestion", async () => {
+        const el = await fixture(
+            html`
+                <aalam-sgn-box>
+                    <div slot="sgn-item-template">{name}</div>
+                </aalam-sgn-box>
+            `
+        );
+        const input = el.shadowRoot?.querySelector("input");
+        input.value = "thi";
+        input.dispatchEvent(
+            new Event("input", { bubbles: true, composed: true })
+        );
+
+        // Simulate a server response or any other source of suggestions
+        const suggestions = [
+            {
+                name: "thiruvarur",
+                description: "My native",
+            },
+        ];
+
+        el.setSuggestion(suggestions, true);
+        await el.updateComplete;
+
+        expect(el.filtered_list.length).to.equal(1);
+        expect(el.filtered_list[0].name).to.equal("thiruvarur");
```

Comment:
Also check the container element. Can also check for load more element here as you setting has_more to true.